voucher/filters.py

```diff
@@ -21,45 +22,75 @@ def apply_date_filter(queryset, value, field, lte):
 
 
 class UseLogFilter(django_filters.FilterSet):
-    user = django_filters.CharFilter(name='wallet__user__username')
     semester = django_filters.CharFilter(name='wallet__semester')
     date_from = django_filters.MethodFilter(action='filter_date_from')
     date_to = django_filters.MethodFilter(action='filter_date_to')
 
-    class Meta:
-        model = UseLog
-        fields = ['id', 'user', 'semester']
-
     def filter_date_from(self, queryset, value):
         return apply_date_filter(queryset, value, 'date_spent', lte=False)
 
     def filter_date_to(self, queryset, value):
         return apply_date_filter(queryset, value, 'date_spent', lte=True)
 
 
+class VoucherUseLogFilter(UseLogFilter):
+    user = django_filters.CharFilter(name='wallet__user__username')
+
+    class Meta:
+        model = VoucherUseLog
+        fields = ['id', 'user', 'semester']
+
+
+class CoffeeUseLogFilter(UseLogFilter):
+    card = django_filters.CharFilter(name='wallet__card__card_uid')
+
+    class Meta:
+        model = CoffeeUseLog
+        fields = ['id', 'card', 'semester']
+
+
 class WalletFilter(django_filters.FilterSet):
-    user = django_filters.CharFilter(name='user__username')
-    card_number = django_filters.MethodFilter(action='filter_card_number')
     valid = django_filters.MethodFilter(action='filter_active')
 
     class Meta:
         model = Wallet
-        fields = ['user', 'card_number', 'semester']
-
-    def filter_card_number(self, queryset, value):
```

Comment:
Okay. Hvis vi går for det så burde vi i så fall kanskje bare fjerne alt som er av koblinger mellom Card og wallets. Hva med å rename Card også, så blir det tydeligere hva det brukes til? F.eks. AccessCard eller noe?